### PR TITLE
Make eye(s) of god NODROP only when equipped

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
tweak: The "eye of god" only becomes unremovable when put in your eye.
/:cl:

- The lavaland chest reward "eye of god", will only become NODROP (ie.
  unremovable) when actually placed in someone's glasses slot, rather
  than becoming immediately stuck in your hand.
- Eyes of god now hurt when implanted, causing temporary blindness and
  screaming.
- A pre-doubled eye of god is now a typepath.
- Eyes of god are no longer made out of glass.
- Glasses colouring procs have been moved to the top of the glasses
  file, but unchanged.

---
## Why It's Good For The Game

Thematically, it makes more sense that the eyes are unremovable only when implanted or "worn", and not that they're just made of magical superglue.
They're made of glass only because the basic glasses type are made of glass, but they're... fleshy eyes cut from a monster.
More screaming and messages gives the item a tiny bit more thematic colour.
A "doubled" type makes it easier for mappers and admins, if they care about such things.

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
👀

**Comment by Fikou** [comment]
i feel like this is even more shitcode than the old version

**Comment by nullAdept** [comment]
Could you also apply this change to the cursed ice hiking boots?

**Comment by coiax** [comment]
@Fikou  Most of the "fiddlyness" comes from getting the grammar of the messages correct for the "painful insertion". Apart from that, isn't the structure about the same?

**Comment by coiax** [comment]
> Could you also apply this change to the cursed ice hiking boots?

#55654

File: code/modules/clothing/glasses/_glasses.dm [hunk 1]
```diff
@@ -434,57 +467,60 @@
...
-		if(H && H.glasses == src)
-			if(old_colour_type)
-				H.remove_client_colour(old_colour_type)
-			if(glass_colour_type)
-				H.update_glasses_color(src, 1)
+	var/verbs = "burrows"
+	if(src.double)
+		verbs = "burrow"
 
+	to_chat(victim, "<span class='userdanger'>You experience blinding pain, as [src] [verbs] into your skull.</span>")
```
Comment (Fikou) [inline code/modules/clothing/glasses/_glasses.dm:512]:
```suggestion
	to_chat(victim, "<span class='userdanger'>You experience blinding pain, as [src] [double ? "burrow" : "burrows"] into your skull.</span>")
```

File: code/modules/clothing/glasses/_glasses.dm [hunk 2]
```diff
@@ -434,57 +467,60 @@
...
-					H.client.prefs.uses_glasses_colour = !H.client.prefs.uses_glasses_colour
-					if(H.client.prefs.uses_glasses_colour)
-						to_chat(H, "<span class='notice'>You will now see glasses colors.</span>")
-					else
-						to_chat(H, "<span class='notice'>You will no longer see glasses colors.</span>")
-					H.update_glasses_color(src, 1)
-	else
-		return ..()
+/obj/item/clothing/glasses/godeye/proc/pain(mob/living/victim)
+	// If pain/pain immunity ever implemented, check for it here.
```
Comment (ATH1909) [inline code/modules/clothing/glasses/_glasses.dm:506]: bruh

File: code/modules/clothing/glasses/_glasses.dm [hunk 3]
```diff
@@ -434,57 +467,60 @@
...
-				C.update_inv_wear_mask()
+				C.update_inv_glasses()
+
+				// Apply pain now, so message is still for solo eye bore
+				if(C.glasses == src)
+					pain(C)
+
+			src.name = initial(T.name)
+			src.desc = initial(T.desc)
+			src.double = TRUE
```
Comment (Fikou) [inline code/modules/clothing/glasses/_glasses.dm:499]:
```suggestion
			name = initial(T.name)
			desc = initial(T.desc)
			double = TRUE
```

File: code/modules/clothing/glasses/_glasses.dm [hunk 4]
```diff
@@ -434,57 +467,60 @@
...
-		if(W.icon_state == "godeye")
-			W.icon_state = "doublegodeye"
-			W.inhand_icon_state = "doublegodeye"
-			W.desc = "A pair of strange eyes, said to have been torn from an omniscient creature that used to roam the wastes. There's no real reason to have two, but that isn't stopping you."
+	if(istype(W, /obj/item/clothing/glasses/godeye) && W != src && W.loc == user)
+		if(!double)
+			var/obj/item/clothing/glasses/godeye/double/T = /obj/item/clothing/glasses/godeye/double
+
+			src.icon_state = initial(T.icon_state)
+			src.inhand_icon_state = initial(T.inhand_icon_state)
```
Comment (Fikou) [inline code/modules/clothing/glasses/_glasses.dm:487]:
```suggestion
			icon_state = initial(T.icon_state)
			inhand_icon_state = initial(T.inhand_icon_state)
```

**Pull request merged**